tests/python/gaia-ui-tests/gaiatest/tests/functional/contacts/test_sort_contacts.py

```diff
@@ -24,20 +22,9 @@ def setUp(self):
         self.sorted_contacts_name_by_last = sorted(self._contacts_name_list, key=lambda name: name[1])
 
     def test_sort_contacts(self):
-        order_by_last_name_default = False
-
         contacts_app = Contacts(self.marionette)
         contacts_app.launch()
 
-        # if "Order by last name" switch is on, turn off it
-        contacts_settings = contacts_app.tap_settings()
-        self.assertEqual(contacts_settings.order_by_last_name, order_by_last_name_default)
-        # This test starts from when order_by_last_name is false
-        if order_by_last_name_default:
-            contacts_settings.tap_order_by_last_name()
-        contacts_settings.tap_done()
-
-        # here we wait for contacts once the correct order has been set
```

Comment:
After a closer look at the test, I think there is some value of not having the default value set at the beginning of the test. It prevents from mental mapping. That is to say:
One day, your test fails at this line:
``` python
toggle_stuff()
```

By just looking at the stacktrace, you have no idea what should be the state of `stuff`. So you go to the code, to check what the test was supposed to do, and you see:
``` python
# more line of code above
if default_stuff_state:
    toggle_stuff()
# more lines there
toggle_stuff()
```

You still don't know what's going on, so you look for that variable, and then you remember that `stuff` should be true. Then you read the test again:
``` python
default_stuff_state = True

if default_stuff_state:
    toggle_stuff()
# ...
toggle_stuff()
```
In which state should `stuff` at the last line? To answer this question, you have to think about it for (at least) 3-4 seconds, and do the conversion each time you toggle `stuff`. This is not trivial.

I agree on the fact that putting the original state is kind of factorization, which is good. But I (personally) think an end-to-end test should be close to English sentences, with no mental mapping. I think this has to be discussed in the best practice document.

PS : I didn't think my comment would be that verbose. Sorry for the ranting.